// @polar-sh/sdk/dist/commonjs/models/operations/customerportalordersretrypayment.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomerPortalOrdersRetryPaymentRequest$ = exports.CustomerPortalOrdersRetryPaymentRequest$outboundSchema = exports.CustomerPortalOrdersRetryPaymentRequest$inboundSchema = exports.CustomerPortalOrdersRetryPaymentSecurity$ = exports.CustomerPortalOrdersRetryPaymentSecurity$outboundSchema = exports.CustomerPortalOrdersRetryPaymentSecurity$inboundSchema = void 0;
exports.customerPortalOrdersRetryPaymentSecurityToJSON = customerPortalOrdersRetryPaymentSecurityToJSON;
exports.customerPortalOrdersRetryPaymentSecurityFromJSON = customerPortalOrdersRetryPaymentSecurityFromJSON;
exports.customerPortalOrdersRetryPaymentRequestToJSON = customerPortalOrdersRetryPaymentRequestToJSON;
exports.customerPortalOrdersRetryPaymentRequestFromJSON = customerPortalOrdersRetryPaymentRequestFromJSON;
const z = __importStar(require("zod"));
const primitives_js_1 = require("../../lib/primitives.js");
const schemas_js_1 = require("../../lib/schemas.js");
/** @internal */
exports.CustomerPortalOrdersRetryPaymentSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
exports.CustomerPortalOrdersRetryPaymentSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return (0, primitives_js_1.remap)(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalOrdersRetryPaymentSecurity$;
(function (CustomerPortalOrdersRetryPaymentSecurity$) {
    /** @deprecated use `CustomerPortalOrdersRetryPaymentSecurity$inboundSchema` instead. */
    CustomerPortalOrdersRetryPaymentSecurity$.inboundSchema = exports.CustomerPortalOrdersRetryPaymentSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersRetryPaymentSecurity$outboundSchema` instead. */
    CustomerPortalOrdersRetryPaymentSecurity$.outboundSchema = exports.CustomerPortalOrdersRetryPaymentSecurity$outboundSchema;
})(CustomerPortalOrdersRetryPaymentSecurity$ || (exports.CustomerPortalOrdersRetryPaymentSecurity$ = CustomerPortalOrdersRetryPaymentSecurity$ = {}));
function customerPortalOrdersRetryPaymentSecurityToJSON(customerPortalOrdersRetryPaymentSecurity) {
    return JSON.stringify(exports.CustomerPortalOrdersRetryPaymentSecurity$outboundSchema.parse(customerPortalOrdersRetryPaymentSecurity));
}
function customerPortalOrdersRetryPaymentSecurityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalOrdersRetryPaymentSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersRetryPaymentSecurity' from JSON`);
}
/** @internal */
exports.CustomerPortalOrdersRetryPaymentRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
exports.CustomerPortalOrdersRetryPaymentRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var CustomerPortalOrdersRetryPaymentRequest$;
(function (CustomerPortalOrdersRetryPaymentRequest$) {
    /** @deprecated use `CustomerPortalOrdersRetryPaymentRequest$inboundSchema` instead. */
    CustomerPortalOrdersRetryPaymentRequest$.inboundSchema = exports.CustomerPortalOrdersRetryPaymentRequest$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersRetryPaymentRequest$outboundSchema` instead. */
    CustomerPortalOrdersRetryPaymentRequest$.outboundSchema = exports.CustomerPortalOrdersRetryPaymentRequest$outboundSchema;
})(CustomerPortalOrdersRetryPaymentRequest$ || (exports.CustomerPortalOrdersRetryPaymentRequest$ = CustomerPortalOrdersRetryPaymentRequest$ = {}));
function customerPortalOrdersRetryPaymentRequestToJSON(customerPortalOrdersRetryPaymentRequest) {
    return JSON.stringify(exports.CustomerPortalOrdersRetryPaymentRequest$outboundSchema.parse(customerPortalOrdersRetryPaymentRequest));
}
function customerPortalOrdersRetryPaymentRequestFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.CustomerPortalOrdersRetryPaymentRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersRetryPaymentRequest' from JSON`);
}
