// @polar-sh/sdk/dist/esm/models/operations/customerportalordersretrypayment.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { remap as remap$ } from "../../lib/primitives.js";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const CustomerPortalOrdersRetryPaymentSecurity$inboundSchema = z.object({
    customer_session: z.string(),
}).transform((v) => {
    return remap$(v, {
        "customer_session": "customerSession",
    });
});
/** @internal */
export const CustomerPortalOrdersRetryPaymentSecurity$outboundSchema = z.object({
    customerSession: z.string(),
}).transform((v) => {
    return remap$(v, {
        customerSession: "customer_session",
    });
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalOrdersRetryPaymentSecurity$;
(function (CustomerPortalOrdersRetryPaymentSecurity$) {
    /** @deprecated use `CustomerPortalOrdersRetryPaymentSecurity$inboundSchema` instead. */
    CustomerPortalOrdersRetryPaymentSecurity$.inboundSchema = CustomerPortalOrdersRetryPaymentSecurity$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersRetryPaymentSecurity$outboundSchema` instead. */
    CustomerPortalOrdersRetryPaymentSecurity$.outboundSchema = CustomerPortalOrdersRetryPaymentSecurity$outboundSchema;
})(CustomerPortalOrdersRetryPaymentSecurity$ || (CustomerPortalOrdersRetryPaymentSecurity$ = {}));
export function customerPortalOrdersRetryPaymentSecurityToJSON(customerPortalOrdersRetryPaymentSecurity) {
    return JSON.stringify(CustomerPortalOrdersRetryPaymentSecurity$outboundSchema.parse(customerPortalOrdersRetryPaymentSecurity));
}
export function customerPortalOrdersRetryPaymentSecurityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalOrdersRetryPaymentSecurity$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersRetryPaymentSecurity' from JSON`);
}
/** @internal */
export const CustomerPortalOrdersRetryPaymentRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const CustomerPortalOrdersRetryPaymentRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var CustomerPortalOrdersRetryPaymentRequest$;
(function (CustomerPortalOrdersRetryPaymentRequest$) {
    /** @deprecated use `CustomerPortalOrdersRetryPaymentRequest$inboundSchema` instead. */
    CustomerPortalOrdersRetryPaymentRequest$.inboundSchema = CustomerPortalOrdersRetryPaymentRequest$inboundSchema;
    /** @deprecated use `CustomerPortalOrdersRetryPaymentRequest$outboundSchema` instead. */
    CustomerPortalOrdersRetryPaymentRequest$.outboundSchema = CustomerPortalOrdersRetryPaymentRequest$outboundSchema;
})(CustomerPortalOrdersRetryPaymentRequest$ || (CustomerPortalOrdersRetryPaymentRequest$ = {}));
export function customerPortalOrdersRetryPaymentRequestToJSON(customerPortalOrdersRetryPaymentRequest) {
    return JSON.stringify(CustomerPortalOrdersRetryPaymentRequest$outboundSchema.parse(customerPortalOrdersRetryPaymentRequest));
}
export function customerPortalOrdersRetryPaymentRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => CustomerPortalOrdersRetryPaymentRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'CustomerPortalOrdersRetryPaymentRequest' from JSON`);
}
